File: Sorting Algo/Basics-Bubble,Insertion,Selection/bubbleSort.js
/*Keep comparing two adjacent values if larger is first then swap and keep swapping until the
largest is moved to the end
o  ...O
Arr->3,5,2,45.23

Step 1. 3,2,5,23,45
Step 2. 2,3,5,23,45

*/


function bubbleSort(arr) {

    for (let i = arr.length - 1; i > 0; i--) {
        for (let j = 0; j < i; j++) {
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j + 1];
                arr[j + 1] = arr[j];
                arr[j] = temp;

            }
            console.log(arr)
        }
    }
    return arr;
}

console.log(bubbleSort([3, 5, 2, 45, 23]))

File: Sorting_1/src/BubbleSort.java

public class BubbleSort {
	public static void printArray(int [] arr) {
		for(int i:arr) {
			System.out.print(i+" ");
		}
	}
	public static void bubbleSort(int [] arr) {
	int n=arr.length;
	int i=0;
	while(i<n-1) {
		int j=0;
		while(j<(n-i-1)) {
			if(arr[j]>arr[j+1]) {
				int temp=arr[j];
				arr[j]=arr[j+1];
				arr[j+1]=temp;
			}
			j++;
		}
		printArray(arr);
		System.out.println();
	i++;
	}
}
	
	public static void main(String[] args) {
		int arr[]= {1,22,3,5,6,7,-1};
		bubbleSort(arr);


	}

}
